user.go

```diff
@@ -37,53 +110,87 @@ func (h UserHandler) AddUserHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	
+	user.ID = primitive.NewObjectID()
```

Comment:
is this necessary ? doesn't mongo create it's own id?

Reply:
Yes because user.ID zero value is an empty string and if I don't set it then _id would always be an empty string, there might be an option to ignore user.ID but from what I saw there isn't.

Reply:
try use `,omitempty`  [link](https://stackoverflow.com/questions/24216510/empty-or-not-required-struct-fields-in-golang)

I believe its a better option than generating id here...

Reply:
I still need to set my own ObjectID here in case someone sets _id in the request's body, I don't think it matters that much whether we generate the ObjectID in our code or in MongoDB.

Reply:
in this case you can just `user.ID = nil` 
it'll solve it.

Reply:
Can't assign nil to this type

Reply:
there is [Undefined](https://godoc.org/github.com/mongodb/mongo-go-driver/bson/primitive#Undefined) type exist in this package...

Reply:
`user.ID = primitive.NilObjectID`

what do you say about it?